src/sentry/static/sentry/app/stores/projectsStore.jsx

```diff
@@ -37,23 +41,29 @@ const ProjectsStore = Reflux.createStore({
 
     this.itemsById = {
       ...this.itemsById,
+      [newProject.id]: newProject,
     };
 
-    this.itemsById[newProject.id] = newProject;
-
     // Ideally we'd always trigger this.itemsById, but following existing patterns
     // so we don't break things
     this.trigger(new Set([prevProject.id]));
   },
 
   onCreateSuccess(project) {
-    this.itemsById[project.id] = project;
+    this.itemsById = {
+      ...this.itemsById,
+      [project.id]: project,
+    };
     this.trigger(new Set([project.id]));
   },
 
   onUpdateSuccess(data) {
     let project = this.getById(data.id);
-    Object.assign(project, data);
+    let newProject = Object.assign({}, project, data);
```

Comment:
I think mutating `project` here could lead to unexpected problems, so create a new object here

Reply:
I think generally it is the entire project object, but at least this way we can support partial updates if needed.

Reply:
The API returns the entire project object for partial updates, so all properties have to be present in data already.

I've been wondering why this Object.assign was ever here originally, and the only reason I could think of was that we actually wanted it to be the same object back then.

Reply:
Yeah I think it's possible that the object mutation was intended, but I've gone and tested and everything seems fine. The only place we really use this is for bookmarking (and now adding/removing teams from projects).

Reply:
We should probably clean this up at some future time then, but for now 👍